changelog.d/10894.feature

```diff
@@ -0,0 +1 @@
+Add a `user_may_send_3pid_invite` spam checker callback for modules to allow or deny 3PID invites.
```

Comment:
Do we want to keep the changelogs?

Reply:
I don't think we care, we don't generate changelogs with towncrier for this repo.